Lut: Refactoring API and clean up docs.

ycw opened the pull request:
examples/jsm/math/Lut.js:
examples/js/math/Lut.js:
- deprecated the `Lut#addColorMap()`; (added `console.warn(..)`)
- added static `Lut.AddColorMap()` 
- cleaned up the `.n`; default value is now defined in single place. 

examples/jsm/math/Lut.d.ts
- added tsdoc for `Lut#addColorMap()` (marked `@deprecated`)
- added `static AddColorMap(..);`

Docs Lut.html:
- removed obsoleted apis, e.g. legend...
- added missing apis, e.g. `.lut[]`, `createCanvas()` ...  
- updated "Code Example" section.
- added "Examples" section; added 1 example; links to "Geometry / Colors / LookupTable".
- added "Statics" section; added 1 func, `Lut.AddColorMap()`
- fixed source link

ycw committed:
update Lut

mrdoob commented on docs/examples/en/math/Lut.html:
```diff
@@ -11,135 +11,149 @@
 
 		<p>
-			[link:https://github.com/mrdoob/three.js/blob/master/examples/jsm/math/[path].js examples/jsm/math/[path].js]
+			[link:https://github.com/mrdoob/three.js/blob/master/examples/jsm/math/Lut.js examples/jsm/math/Lut.js]
```
Hmm, this needs to be fixed somewhere else. Better to leave it as it is for this PR.

ycw replied:
https://github.com/mrdoob/three.js/blob/4d875bbd8f05be5eae13366b06c03e7aaae79cec/docs/page.js#L45

mrdoob commented:
I wonder if we even need `AddColorMap()`.

Why not just...

```js
const colorMap = [
  [ 0.0, 0x1019f7 ],
  [ 0.2, 0x0011e1 ],
  [ 0.4, 0x115cca ],
  [ 0.6, 0x33a5c5 ],
  [ 0.8, 0x25cea5 ],
  [ 1.0, 0x16f782 ]
];

const lut = new Lut( colorMap, 512 );
const color = lut.getColor( 0.5 );
```

ycw commented:
It doesn't even need to expose `ColorMapKeywords` & relative apis (i.e. `addColorMap`)

ycw committed:
refactor Lut

ycw commented:
Refactor

- removed `.set()`. Use `.copy()` instead
- renamed `.lut[]` to `.colors[]`. 
- removed `.n`. This is exactly the `.colors.length`. Less noise.
- refactor: `.setColorMap()`. It accepts string OR array now. 
  - https://github.com/mrdoob/three.js/pull/19561#issuecomment-639177786 
- refactor: make `ColorMapKeywords` dictionary private.
- removed `.addColorMap()`. 

Docs

- updated "Code Example". Added snippet to show new ctor usage
- cleanup. 

Types

- updated to use precise type alias.

Mugen87 commented:
For testing: https://rawcdn.githack.com/ycw/three.js/dev/examples/webgl_geometry_colors_lookuptable.html

Mugen87 commented:
Unfortunately, this PR is outdated since many things have changed over time. The example code is now based on ES6, TS declaration files have been removed and #22148 fixed the outdated state of the `Lut` documentation.

If a new/different `Lut` API is wanted, it's better to open a new PR.
